pkg/emitter/nats_emitter.go

```diff
@@ -0,0 +1,121 @@
+//
+// Copyright 2022 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package emitter
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+
+	"github.com/guacsec/guac/pkg/handler/processor"
+	"github.com/guacsec/guac/pkg/logging"
+	"github.com/nats-io/nats.go"
+)
+
+// NATS stream
+const (
+	NatsName                string = "GUAC"
+	StreamName              string = "DOCUMENTS"
+	StreamSubjects          string = "DOCUMENTS.*"
+	SubjectNameDocCollected string = "DOCUMENTS.collected"
+	SubjectNameDocProcessed string = "DOCUMENTS.processed"
+	SubjectNameDocParsed    string = "DOCUMENTS.parsed"
+)
+
+var (
+	nc *nats.Conn
+	js nats.JetStreamContext
+)
+
+func JetStreamInit(ctx context.Context, url string, creds string, nkeyFile string, insecure bool) nats.JetStreamContext {
```

Comment:
nit: can have a config struct and pass it in, and put comments in the struct on what fields mean

Reply:
done.